hathor/verification/block_verifier.py

```diff
@@ -107,3 +117,13 @@ def verify_outputs(self, block: BaseTransaction) -> None:
     def verify_data(self, block: Block) -> None:
         if len(block.data) > self._settings.BLOCK_DATA_MAX_SIZE:
             raise TransactionDataError('block data has {} bytes'.format(len(block.data)))
+
+    def verify_mandatory_signaling(self, block: Block) -> None:
+        """Verify whether this block is missing mandatory signaling for any feature."""
+        if not self._settings.FEATURE_ACTIVATION.enable_usage:
+            return
+
+        assert self._feature_service is not None
+
+        if not self._feature_service.is_signaling_mandatory_features(block):
+            raise BlockMustSignalError()
```

Comment:
Should we add information about which feature failed?

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/785/commits/c07032019b78c1ce3cbcaa067d1d65748c337353. Instead of using something like a `(success: bool, error: Optional[str])` tuple, I used the sum types pattern that we discussed the other day. Check out what you think.

Reply:
Nice. Thanks!